File: Task1/php/processdata.php
<!DOCTYPE html>
<html>
<head>
	<meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
	<title>Process Data in PHP</title>
</head>
<body>
	<?php
	if (isset($_GET['username_get']) && isset($_GET['userpassword_get']) && isset($_GET['magicnumber_get'])){
		$num = $_GET['magicnumber_get'];
		$name = $_GET['username_get'];
		$pw = $_GET['userpassword_get'];
		
		for($k = 1; $k<=$num ; $k++){
		$res = "Hello ".$name." with a password of ".$pw ."!";
		echo $res;
		?>
		<br/>
		<?php	
		}
	}
	else{
		if (isset($_POST['username_post']) && isset($_POST['userpassword_post']) && isset($_POST['magicnumber_post'])){
		$num = $_POST['magicnumber_post'];
		$name = $_POST['username_post'];
		$pw = $_POST['userpassword_post'];
		
			for($k = 1; $k<=$num ; $k++){
			$res = "Hello ".$name." with a password of ".$pw ."!";
			echo $res;
			?>
			<br/>
			<?php	
			}
		}
		else{
			echo "The form has not been filled up properly.";
		}
	}
	
?>
</body>
</html>

File: Task1/java/ProcessData.java
import java.util.*;
import java.io.*;

class ProcessData{

  static void toHTML(String content){
    String html = String.format("<!doctype html><html><body><head><title>Process Data Java</title></head>%s</body></html>",content);
    System.out.print("Content-Type: text/html\n\n");
    System.out.println(html);
  }

  public static void main(String[] args){

    Map<String, String> formData = new HashMap <String, String>();
    Map<String, String> env = System.getenv();
    String data = "";
    String htmlContent = "";

    if(env.get("REQUEST_METHOD").toString().equals("GET")){
       data = env.get("QUERY_STRING");
    }
    else if(env.get("REQUEST_METHOD").toString().equals("POST")){
    
      DataInput postOutput = new DataInputStream(System.in);
      String outputLine;
      try{
      while((outputLine = postOutput.readLine()) != null){
        data = data + outputLine;
      }
    }catch(IOException i){
      toHTML("<h1> Sorry, there was an error parsing post data </h1>");
    }
    }
    else{
      toHTML("<h1>There was no GET or POST request submitted through a form!</h1>");
      return;
    }

    String[] formField = data.split("&");

    for(int i = 0; i < formField.length; i++){
      String[] fields = formField[i].split("=");
      formData.put(fields[0],fields[1]);
    }
    int magicNum;
    try{
      magicNum = Integer.parseInt(formData.get("magicnumber"));
    }catch(NumberFormatException e){
      toHTML("<h1> Sorry, There was an error parsing the magic number </h1>");
      return;
    }

    for(int i = 0; i < magicNum; i++){
      htmlContent = htmlContent + ("<h1> Hello " + formData.get("username") + " with a password of " + formData.get("password") + " </h1>");
    }
    toHTML(htmlContent);

  }
}
